firestore/CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This should be included from the top level CMakeLists file
+set(FIRESTORE_CLIENT_VERSION_MAJOR 0)
+set(FIRESTORE_CLIENT_VERSION_MINOR 0)
+set(FIRESTORE_CLIENT_VERSION_PATCH 1)
+
+# Configure the Compiler options, we will be using C++11 features.
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+set(DOXYGEN_PROJECT_NAME "Google Cloud Firestore C++ Client")
+set(DOXYGEN_PROJECT_BRIEF "A C++ Client Library for Google Cloud Firestore")
+set(DOXYGEN_PROJECT_NUMBER
+    "${FIRESTORE_CLIENT_VERSION_MAJOR}.${FIRESTORE_CLIENT_VERSION_MINOR}.${FIRESTORE_CLIENT_VERSION_PATCH}")
+set(DOXYGEN_EXAMPLE_PATH examples)
+set(DOXYGEN_PREDEFINED "FIRESTORE_CLIENT_NS=v${FIRESTORE_CLIENT_VERSION_MAJOR}")
+set(DOXYGEN_EXCLUDE_PATTERNS "*/client/internal/*;*/client/testing/*;*/*_test.cc")
+
+include(GoogleCloudCppCommon)
+
+# Define an interface library, i.e., a library that really has no sources, and
+# add public target options to it.  The targets then use the library via
+# target_link_libraries(target PRIVATE firestore_common_options), which
+# implicitly adds the options to those targets, but does *not* add the options
+# to any targets that depend on them.  This is a CMake idiom to add options to
+# targets without having to painfully set special variables, it also allows
+# users to add more common options without having to touch the code.
+add_library(firestore_common_options INTERFACE)
+    google_cloud_cpp_add_common_options(firestore_common_options)
+
+# Enable unit tests
+enable_testing()
+
+# All tests get added to this target below.
+add_subdirectory(client)
+
+# Install the libraries and headers in the locations determined by
+# GNUInstallDirs
+install(TARGETS 
+    EXPORT firestore-targets
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+# Install proto generated files into firestore/google.
+install(DIRECTORY ${CMAKE_BINARY_DIR}/firestore/google/ DESTINATION include/google FILES_MATCHING PATTERN "*.h")
```

Comment:
You can remove this line I think, since we are not generating any code from the protos (yet).